### DISCUSSION TITLE
Does ui.timer block the UI?

### DISCUSSION THREAD

**Discussion by thompsm** [zauberzeug/nicegui#622] [Q&A] [answered]
Hi,

I'm using a ui.timer to callback a function that updates a ui.row every 10 seconds. Inside the ui.row are some buttons, but it seems they do not respond to mouse clicks while the ui.timer is running? If this is the case, is there a way to make the ui.timer not block the UI?

**Comment by falkoschindler** [comment, marked as answer]
Hi, @thompsm! The main purpose of the `ui.timer` is to _not_ block the UI.
But the callback you are passing the the timer should not block either. So it the callback is blocking for a significant amount of time, the UI will not respond.

Please see this discussion for a similar problem: https://github.com/zauberzeug/nicegui/discussions/590

**Reply by thompsm** [reply]
Hi @falkoschindler. Many thanks for the advice. You're correct that my callback was taking a few seconds and blocking the UI. I've now wrapped the slowest part of the function in an executor and it's much better now.

Also thank you for your fast responses to my problems. I've been trying to create a dashboard type app for a while and started with Dash, and then Streamlit, but neither fit my needs very well. Then I found NiceGUI and it's working out great so far.
